src/app/introduction/activity-1/activity-1.component.ts

```diff
@@ -69,8 +70,9 @@ export class HighlightActivityComponent implements OnInit {
                 });
                 this._sharedData.customSuccessAlert(this.alerts.activitySuccessMsg, this.alerts.activitySuccessTitle);
                 this.activityComplete = true;
-                if (!this.completed) { const currStage = 1;
-                  this._performanceService.openDialog(currStage); }
+          if (!this.completed) { const currStage = 1;
+                  this._performanceService.openDialog(currStage);
+                  this._leaderBoardService.updateLeaderBoard({activity: 'highlightDefinition', level: 'level1'})}
```

Comment:
Does the level attribute serve any purpose?

Reply:
@Raxerz  Ya, it serves a purpose. If you see **PointsConfig.ts** I have declared points to be given to activities as per levels. Currently, as all activities have one level, so we are giving *level1* flag. For multiple levels, we can have *level2*, *level3* flags respectively.

Reply:
Is this being stored in the database?

Reply:
Currently, it's like this.
His score in a particular level is only going to be added when he completes that level for the first time. (not for activities which involve dynamic scores like Odd One Out etc.). So, passing the level flag from the game would ensure that his score is updated for that level.